src/components/entity/ha-entity-picker.ts

```diff
@@ -260,6 +262,30 @@ export class HaEntityPicker extends LitElement {
     }
   }
 
+  private _filterChanged(ev): void {
+    const filterString = ev.detail.value.toLowerCase();
+
+    const states = this._getStates(
+      this._opened,
+      this.hass,
+      this.includeDomains,
+      this.excludeDomains,
+      this.entityFilter,
+      this.includeDeviceClasses
+    );
```

Comment:
We should not get the states of every key press, save the states from line 178 and reuse that. Even though it is memoized, `hass` will probably change and we don't want it to update as long as it is open.